lib/sendwithus.js

```diff
@@ -98,6 +99,7 @@ Sendwithus.prototype._fetch = function () {
 
 Sendwithus.prototype._handleResponse = function (result, response, callback) {
   if (result instanceof Error) {
+    this.emit("response", response.status, result, response);
```

Comment:
@rob-swu something like this? Emit a response even when an error happens?

Reply:
no, we need the actual callback function to be called

since it never reaches the failure states in the `then` and will never call the callback on fail when we call handleResponse in the `then`. The user provides the callback function so we need to call it. 

so either we could just call the handleResponse function directly in the 
```
.catch((err) => this._handleResponse(err, undefined, callback)
```

or we could pull the part of handleResponse out that deals with the errors into its own function and call it, so we don't have an undefined parameter

Reply:
updated to move both the `then()` and `catch() `into the higher `Sendwithus.prototype._fetch`